pkg/modulewriter/packerwriter.go

```diff
@@ -58,36 +59,45 @@ func writePackerAutovars(vars map[string]cty.Value, dst string) error {
 // writeDeploymentGroup writes any needed files to the top and module levels
 // of the blueprint
 func (w PackerWriter) writeDeploymentGroup(
-	depGroup config.DeploymentGroup,
-	globalVars map[string]interface{},
+	dc config.DeploymentConfig,
+	grpIdx int,
 	deployDir string,
-) error {
-	ctyGlobals, err := config.ConvertMapToCty(globalVars)
+) (groupMetadata, error) {
+	ctyGlobals, err := config.ConvertMapToCty(dc.Config.Vars)
 	if err != nil {
-		return fmt.Errorf(
+		return groupMetadata{}, fmt.Errorf(
 			"error converting deployment vars to cty for writing: %w", err)
 	}
+
+	depGroup := dc.Config.DeploymentGroups[grpIdx]
 	groupPath := filepath.Join(deployDir, depGroup.Name)
+	allInputs := []string{}
 	for _, mod := range depGroup.Modules {
 
 		ctySettings, err := config.ConvertMapToCty(mod.Settings)
+		allInputs = append(allInputs, orderKeys(ctySettings)...)
```

Comment:
Is there a chance for collision? Is it possible to have multiple modules in packer group?

Reply:
When enabling IGC, we plan to impose a restriction to have a single module limit for Packer groups, but nevertheless it would be safer to do it right so I've changed to the `map[string]bool` approach used elsewhere.